common:
  img_topic: "/camera/color/image_raw/compressed"
  lid_topic: "/ouster/points"
  imu_topic: "/imu/data"
  is_compressed_image: true
  dynamic_slam: true
  mask_topic: ""
  img_en: 1             # Enable visual mode
  lidar_en: 1           # Enable lidar mode
  ros_driver_bug_fix: false

extrin_calib:
  # Hilti-2022
  extrinsic_T: [0, 0, 0] 
  extrinsic_R: [ 1, 0, 0, 
                 0, 1, 0, 
                 0, 0, 1]
  

  Rcl: [ 0,  0, -1,
         0, 1, 0,
         1, 0, 0]
  Pcl: [0, 0, 0 ]

time_offset:
  imu_time_offset: 0.0
  img_time_offset: 0.0
  exposure_time_init: 0.0

preprocess:
  point_filter_num: 3
  filter_size_surf: 0.1
  lidar_type: 3 # Ouster
  scan_line: 32
  blind: 0.5
 
vio:
  max_iterations: 5
  outlier_threshold: 500
  img_point_cov: 1000
  patch_size: 8
  patch_pyrimid_level: 4
  normal_en: true
  raycast_en: false
  inverse_composition_en: false
  exposure_estimate_en: true
  inv_expo_cov: 0.1

imu:
  imu_en: true
  imu_int_frame: 30
  acc_cov: 0.5 # 0.1
  gyr_cov: 0.01
  b_acc_cov: 0.0001 # 0.1
  b_gyr_cov: 0.0001 # 0.1

lio:
  max_iterations: 5
  dept_err: 0.02
  beam_err: 0.05
  min_eigen_value: 0.0001 # 0.0025
  voxel_size: 0.4
  max_layer: 2
  max_points_num: 100
  layer_init_num: [5, 5, 5, 5, 5]

local_map:  
  map_sliding_en: false
  half_map_size: 100
  sliding_thresh: 8

uav:
  imu_rate_odom: false
  gravity_align_en: false

publish:
  dense_map_en: true
  pub_effect_point_en: false 
  pub_plane_en: false
  pub_scan_num: 1
  blind_rgb_points: 0.0

evo:
  seq_name: "exp09_cupola"
  pose_output_en: true

pcd_save:
  pcd_save_en: false
  colmap_output_en: false # need to set interval = -1
  filter_size_pcd: 0.15
  interval: -1
  # how many LiDAR frames saved in each pcd file;
  # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
